packages/ra-supabase-ui-materialui/src/guessers/CreateGuesser.tsx

```diff
@@ -0,0 +1,131 @@
+import { capitalize, singularize } from 'inflection';
+
+import { inferElementFromType } from './inferElementFromType';
+
+export const CreateGuesser = (props: CreateProps & { enableLog?: boolean }) => {
+    const {
+        mutationOptions,
+        resource,
+        record,
+        transform,
+        redirect,
+        disableAuthentication,
+        ...rest
+    } = props;
+    return (
+        <CreateBase
+            resource={resource}
+            record={record}
+            redirect={redirect}
+            transform={transform}
+            mutationOptions={mutationOptions}
+            disableAuthentication={disableAuthentication}
+        >
+            <CreateGuesserView {...rest} />
+        </CreateBase>
+    );
+};
+
+export const CreateGuesserView = (
+    props: CreateViewProps & {
+        enableLog?: boolean;
+    }
+) => {
+    const { data: schema, error, isPending } = useAPISchema();
+    const resource = useResourceContext();
+    const [child, setChild] = React.useState<ReactNode>(null);
+    if (!resource) {
+        throw new Error('CreateGuesser must be used withing a ResourceContext');
+    }
+    const { enableLog = process.env.NODE_ENV === 'development', ...rest } =
+        props;
+
+    React.useEffect(() => {
+        if (isPending || error) {
+            return;
+        }
+        const resourceDefinition = schema.definitions?.[resource];
+        const requiredFields = resourceDefinition?.required || [];
+        if (!resourceDefinition || !resourceDefinition.properties) {
+            throw new Error(
+                `The resource ${resource} is not defined in the API schema`
+            );
+        }
+        if (!resourceDefinition || !resourceDefinition.properties) {
+            return;
+        }
+        const inferredInputs = Object.keys(resourceDefinition.properties)
+            .filter((source: string) => source !== 'id')
+            .filter(
+                source =>
+                    resourceDefinition.properties![source].format !== 'tsvector'
+            )
+            .map((source: string) =>
+                inferElementFromType({
+                    name: source,
+                    types: editFieldTypes,
+                    description:
+                        resourceDefinition.properties![source].description,
+                    format: resourceDefinition.properties![source].format,
+                    type: (resourceDefinition.properties &&
+                    resourceDefinition.properties[source] &&
+                    typeof resourceDefinition.properties[source].type ===
+                        'string'
+                        ? resourceDefinition.properties![source].type
+                        : 'string') as string,
+                    requiredFields,
+                })
+            );
+        const inferredForm = new InferredElement(
+            editFieldTypes.form,
+            null,
+            inferredInputs
+        );
```

Comment:
Sorry to insist but this could be extracted at least as a `getInferredView` function taking the following parameters:
- `types`: here it would be `editFieldTypes`, `listFieldTypes` in `ListGuesser`
- `containerType`: here `editFieldTypes.form`, `listFieldTypes.table` in `ListGuesser`

That would make it a lot easier to read and avoid a lot of duplication. It could also be a hook but that may hurt comprehension of the code